### PR TITLE
Fix SwiftUI.BPKShimmer not working in iOS 17

### PR DESCRIPTION
**Pull request by iHandle**
**Changes**
 Fix SwiftUI.BPKShimmer not working in iOS 17

**Preview**

https://github.com/Skyscanner/backpack-ios/assets/16792577/4c8fa30b-233b-4d85-94cd-5aa1c8b5a2b4

https://github.com/Skyscanner/backpack-ios/assets/16792577/c3291b3b-756a-42b7-bcec-85242b6daa8a

+ [x] Check this if you have read and followed the [contributing guidelines](https://github.com/Skyscanner/backpack-ios/blob/main/CONTRIBUTING.md)

Remember to include the following changes:
+ [ ] `README.md`
+ [ ] Tests
+ [ ] [Screenshotting code](https://github.com/Skyscanner/backpack-ios/blob/main/Example/Backpack%20Screenshot/Screenshots.swift)
+ [ ] Adding a component? Remember to expose it in the [main `Backpack.h` header file](https://github.com/Skyscanner/backpack-ios/tree/main/Backpack/Backpack.h)

_If you are curious about how we review, please read through the [code review guidelines](https://github.com/Skyscanner/backpack/blob/main/CODE_REVIEW_GUIDELINES.md)_

### CODE REVIEW

File: Backpack-SwiftUI/Skeleton/Classes/BPKShimmer.swift [hunk 1]
```diff
@@ -19,49 +19,33 @@
 import SwiftUI
 
 struct BPKShimmer: ViewModifier {
-    @State private var phase: CGFloat = 0
+    @State private var offset = 0.0
     private let duration = 1.0
-    private let bounce = false
-    private let delay = 0.2
```
Comment (iHandle) [inline Backpack-SwiftUI/Skeleton/Classes/BPKShimmer.swift:25]: The `dealy` property was never used before

File: Backpack-SwiftUI/Skeleton/Classes/BPKShimmer.swift [hunk 2]
```diff
@@ -19,49 +19,34 @@
...
-            ))
-            .onAppear { phase = 0.8 }
+            .mask(linearGradient)
+            .animation(linearAnimation, value: offset)
+            .onAppear {
+                offset = 1.0 + bandSize
+            }
     }
-
-    struct ShimmerMask: AnimatableModifier {
```
Comment (iHandle) [inline Backpack-SwiftUI/Skeleton/Classes/BPKShimmer.swift:36]: `AnimatableModifier` is deprecated: https://developer.apple.com/documentation/swiftui/animatablemodifier

File: Backpack-SwiftUI/Skeleton/Classes/BPKShimmer.swift [hunk 3]
```diff
@@ -19,49 +19,34 @@
 import SwiftUI
 
 struct BPKShimmer: ViewModifier {
-    @State private var phase: CGFloat = 0
+    @State private var offset = 0.0
     private let duration = 1.0
     private let bounce = false
-    private let delay = 0.2
```
Comment (iHandle) [inline Backpack-SwiftUI/Skeleton/Classes/BPKShimmer.swift:25]: The dealy property was never used before

Comment (frugoman) [reply Backpack-SwiftUI/Skeleton/Classes/BPKShimmer.swift:25]: 👍

File: Backpack-SwiftUI/Skeleton/Classes/BPKShimmer.swift (hunk 2 above)
Comment (frugoman) [reply Backpack-SwiftUI/Skeleton/Classes/BPKShimmer.swift:36]: Nice one!

**Review by frugoman — APPROVED** [review]
Looks very good!

**Pull request merged**